types/react-scroll-to-bottom/index.d.ts

```diff
@@ -0,0 +1,208 @@
+         * Set the class name for the follow button
+         */
+        followButtonClassName?: string;
+        /**
+         * Set the initial scroll behavior, either `"auto"` (discrete scrolling) or `"smooth"`
+         */
+        initialScrollBehavior?: 'auto' | 'smooth';
+        /**
+         * Set it to `"bottom"` for scroll-to-bottom, `"top"` for scroll-to-top
+         */
+        mode?: 'bottom' | 'top';
+        /**
+         * Set the nonce for Content Security Policy
+         */
+        nonce?: string;
+        /**
+         * A function to determine how far should scroll when scroll is needed
+         */
+        scroller?: (values: {
+            /**
+             * Maximum distance (in pixel) to scroll
+             */
+            maxValue: number;
+            /**
+             * Minimum distance (in pixel) to scroll
+             */
+            minValue: number;
+            /**
+             * View height of the scrollable container
+             */
+            offsetHeight: number;
+            /**
+             * Total height of the content in the container, must be equal or greater than offsetHeight
+             */
+            scrollHeight: number;
+            /**
+             * Current scroll position (in pixel)
+             */
+            scrollTop: number;
+        }) => number;
+        /**
+         * Set the class name for the container element that house all `props.children`
+         */
+        scrollViewClassName?: string;
+    }>,
+): null;
+
+/**
+ * Scroll panel to specified position
+ */
+export function useScrollTo(): (scrollTop: number | '100%') => void;
+
+/**
+ * Scroll panel to bottom
+ */
+export function useScrollToBottom(): () => void;
+
+/**
+ * Scroll panel to end (depends on `mode`)
+ */
+export function useScrollToEnd(): () => void;
+
+/**
+ * Scroll panel to start (depends on `mode`)
+ */
+export function useScrollToStart(): () => void;
+
+/**
+ * Scroll panel to top
+ */
+export function useScrollToTop(): () => void;
+
+/**
+ * Observe scroll position change by passing a callback function
+ */
+export function useObserveScrollPosition(observer: (({ scrollTop }: { scrollTop: number }) => void) | false): void;
+
+/**
+ * `true` if the panel is animating scroll effect
+ */
+export function useAnimating(): [boolean];
+
+/**
+ * `true` if the panel is animating scroll effect and towards the end (depends on `mode`)
+ */
+export function useAnimatingToEnd(): [boolean];
+
+/**
+ * `true` if the panel is currently near bottom
+ */
+export function useAtBottom(): [boolean];
+
+/**
+ * `true` if the panel is currently near the end (depends on `mode`)
+ */
+export function useAtEnd(): [boolean];
+
+/**
+ * `true` if the panel is currently near the start (depends on `mode`)
+ */
+export function useAtStart(): [boolean];
+
+/**
+ * `true` if the panel is currently near top
+ */
+export function useAtTop(): [boolean];
+
+/**
+ * 	`"bottom"` for scroll-to-bottom, `"top"` for scroll-to-top
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/react-scroll-to-bottom/index.d.ts;
  Line: 136;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.